source/common/stream_info/stream_info_impl.h

```diff
@@ -20,12 +21,14 @@ struct StreamInfoImpl : public StreamInfo {
   StreamInfoImpl(TimeSource& time_source)
       : time_source_(time_source), start_time_(time_source.systemTime()),
         start_time_monotonic_(time_source.monotonicTime()),
-        filter_state_(std::make_shared<FilterStateImpl>(FilterState::LifeSpan::FilterChain)) {}
+        filter_state_(std::make_shared<FilterStateImpl>(FilterState::LifeSpan::FilterChain)),
+        request_id_extension_(Http::RequestIDExtensionFactory::noopInstance()) {}
```

Comment:
looking at this, I think these constructors can mostly be collapsed with each one invoking the previous with fewer arguments. This would be a nice follow up.

Reply:
Sounds good, will send a follow up pr. Does it need a TODO for the time being?

Reply:
Created #10691, will rebase once this one is merged.